GreyNoise-Canary-Threat-Intel-Report.sh

```diff
@@ -4,18 +4,16 @@
 # Canary support will be happy to enable API access to your console. See the README for further information.
 
 # Replace these placeholder values with your own
-export CANARY_HASH="abcd1234"
-export CANARY_TOKEN="11a2222bb3333c444d555e6f777ggg88"
+CANARY_HASH="abcd1234"
+CANARY_TOKEN="11a2222bb3333c444d555e6f777ggg88"
+CANARY_IP_LIST="canary-ips-$(date +%Y-%m-%d).txt"
 
 # Query Canary API for all events, extract IPs, remove double quotes, and write to a text file for GreyNoise
-curl https://$CANARY_HASH.canary.tools/api/v1/incidents/outside_bird/search -d auth_token=$CANARY_TOKEN -d node_ids="INSERT_NODE_IDS" -G | jq '.src_ips[] | .ip_address' | sed 's/\"//g' | cat > canary-ips-$(date +%Y-%m-%d).txt
+curl "https://${CANARY_HASH}.canary.tools/api/v1/incidents/outside_bird/search -d auth_token=${CANARY_TOKEN}" -d node_ids="INSERT_NODE_IDS" -G \
+    | jq --raw-output '.src_ips[] | .ip_address' > "${CANARY_IP_LIST}"
 
 # Read through each line of the text file, send each IP to the GreyNoise Community API for context, and write the results to a JSON file
-file="canary-ips-$(date +%Y-%m-%d).txt"
-lines=$(cat $file)
-
-for line in $lines
+while read -r ip_address
 do
-    curl "https://api.greynoise.io/v3/community/$line" | jq '.' | cat >> GreyNoise-Canary-Threat-Intel-Report-$(date +%Y-%m-%d).json
-done
-
+    curl "https://api.greynoise.io/v3/community/${ip_address}" | jq '.'
+done < "${CANARY_IP_LIST}" > "GreyNoise-Canary-Threat-Intel-Report-$(date +%Y-%m-%d).json"
```

Comment:
This is how one reads lines from a file - Please consider using https://github.com/koalaman/shellcheck